src/main/scala/uclid/ConcreteSimulator.scala

```
@@ -0,0 +1,1293 @@
+                                                                // List of concrete indices expressions 
+                                                                var old_map = ca.value // old array 
+                                                                old_map(eval_indices) = operandValue_1
+                                                                val new_arr = ConcreteArray(old_map)
+                                                                varMap(id) = new_arr
+                                                            }
+                                                            case _ => 
+                                                                throw new Error(f"Attempting ArraySelect on a non-array object ${id.name}")
+                                                        }
+
+                                                    }
+                                                    case _ =>{}
+                                                }
+                                            }
+                                            case _=>{
+                                                throw new Error("Wait for support of Expr "+expr.toString)
+                                            }
+                                        }
+                                    }
+                                    // add support for array
+                                    case _ => {
+                                        throw new Error("Wait for support of Expr "+expr.toString)
+                                        //updateVar(operand_0,evaluate_expr(this,operand_1))
+                                    }
+                                }    
+                            }
+                            //TODO:
+                            //wait for better plan
+                            case InequalityOp() => {
+                                // throw new Error("Unimplemented")
```

Please throw the error